src/core/utils/CorePlugin.ts

```diff
@@ -43,17 +43,27 @@ export class CorePlugin extends JWPlugin {
         // TODO: this is a stub
         const range = this.editor.vDocument.range;
         if (range.isCollapsed()) {
-            range.setStart(range.start.previous());
+            const previous = range.start.previousLeaf();
+            if (previous && range.start.index > 0) {
+                range.setStart(previous, RelativePosition.BEFORE);
+            } else if (previous) {
+                range.setStart(previous, RelativePosition.AFTER);
+            }
```

Comment:
I think I would have done:
```
const previousSibling = range.start.previousSibling();
let previous: VNode;
let position: RelativePosition;
if (previousSibling) {
    previous = previousSibling.firstLeaf();
    position = RelativePosition.BEFORE;
} else {
    previous = range.start.previousLeaf();
    position = RelativePosition.AFTER;
}
```
Not sure that's better though... I think we have a problem here. I think your version is what we want because it is shorter and more generic, in the sense that `previousLeaf` already does the sibling check I rewrote here. However, it requires to check the `range.start.index`, which is not obvious if you don't precisely know what `previousLeaf` does. Maybe we need something like `previousAnchor` or `previousLocation` which would return a tuple `[VNode, RelativePosition]` ? Because whichever of the two above solution we choose, we are inevitably duplicating the `previous` logic in some way.